tests/core/continuous_barycenter_test.py

```diff
@@ -98,11 +94,89 @@ def test_euclidean_barycenter(
     self.assertTrue(jnp.isclose(costs[-2], costs[-1], rtol=threshold))
 
     # Check barycenter has all points roughly in [1,2]^4.
-    # (this is because sampled points where equally set in either [0,1]^4
+    # (this is because sampled points were equally set in either [0,1]^4
     # or [2,3]^4)
     self.assertTrue(jnp.all(out.x.ravel() < 2.3))
     self.assertTrue(jnp.all(out.x.ravel() > .7))
 
+  @parameterized.product(
+      lse_mode=[True, False], epsilon=[1e-1, 5e-1], jit=[True, False]
+  )
+  def test_bures_barycenter(self, lse_mode, epsilon, jit):
+    num_measures = 2
+    num_components = 2
+    dimension = 2
+    bar_size = 2
+    barycentric_weights = jnp.asarray([0.5, 0.5])
+    bures_cost = costs.Bures(dimension=dimension)
+
+    means1 = jnp.array([[-1., 1.], [-1., -1.]])
+    means2 = jnp.array([[1., 1.], [1., -1.]])
+    sigma = 0.01
+    covs1 = sigma * jnp.asarray([
+        jnp.eye(dimension) for i in range(num_components)
+    ])
+    covs2 = sigma * jnp.asarray([
+        jnp.eye(dimension) for i in range(num_components)
+    ])
+
+    y1 = bures_cost.means_and_covs_to_x(means1, covs1)
+    y2 = bures_cost.means_and_covs_to_x(means2, covs2)
+
+    b1 = b2 = jnp.ones(num_components) / num_components
+
+    y = jnp.concatenate((y1, y2))
+    b = jnp.concatenate((b1, b2))
+
+    key = jax.random.PRNGKey(0)
+    keys = jax.random.split(key, num=2)
+    x_init_means = jax.random.uniform(keys[0], (bar_size, dimension))
+    x_init_covs = jax.vmap(
+        lambda a: a @ jnp.transpose(a), in_axes=0
+    )(
+        jax.random.uniform(keys[1], (bar_size, dimension, dimension))
+    )
+
+    x_init = bures_cost.means_and_covs_to_x(x_init_means, x_init_covs)
+
+    bar_p = bar_problems.BarycenterProblem(
+        y,
+        b,
+        weights=barycentric_weights,
+        num_per_segment=jnp.asarray([num_components, num_components]),
+        num_segments=num_measures,
+        max_measure_size=num_components,
+        cost_fn=bures_cost,
+        epsilon=epsilon
+    )
+
+    solver = continuous_barycenter.WassersteinBarycenter(
+        lse_mode=lse_mode, jit=jit
+    )
+
+    out = solver(bar_p, bar_size=bar_size, x_init=x_init)
+    barycenter = out.x
+
+    means_bary, covs_bary = bures_cost.x_to_mean_and_cov(barycenter)
+
+    self.assertTrue(
+        jnp.allclose(
+            means_bary,
```

Comment:
since this should be permutation invariant maybe first compute the mininum of 

jnp.sum( mean_bary- jnp.array([[0., 1.], [0., -1.]]),
jnp.sum( mean_bary- jnp.array([[0., -1.], [0., 1.]])

or even do an OT problem haha :)